src/axom/core/ArrayBase.hpp

```diff
@@ -360,6 +490,28 @@ class ArrayBase<T, 1, ArrayType>
     assert(inBounds(idx));
     return asDerived().data()[idx];
   }
+
+  /*!
+   * \brief Accessor, returns a reference to the given value.
+   * For multidimensional arrays, indexes into the (flat) raw data.
+   *
+   * \param [in] idx the position of the value to return.
+   *
+   * \note equivalent to *(array.data() + idx).
+   *
+   * \pre 0 <= idx < m_num_elements
+   */
+  AXOM_HOST_DEVICE T& flatIdx(const IndexType idx)
```

Comment:
I'm pretty sure we more often use the fully spelled out word `index`.  This should be `flatIndex` especially for user facing symbols.